proof/append.go

```diff
@@ -160,6 +160,47 @@ func CreateTransitionProof(prevOut wire.OutPoint,
 		TapSiblingPreimage: params.TapscriptSibling,
 	}
 
+	if proof.Asset.IsTransferRoot() {
+		stxoInclusionProofs := make(
+			map[asset.SerializedKey]commitment.Proof,
+			len(proof.Asset.PrevWitnesses),
+		)
+		for _, wit := range proof.Asset.PrevWitnesses {
+			if wit.PrevID == nil {
+				return nil, fmt.Errorf("witness %v has no "+
+					"prevID", wit)
+			}
+
+			spentAsset := &asset.Asset{
+				ScriptKey: asset.NewScriptKey(
+					asset.DeriveBurnKey(*wit.PrevID),
+				),
+			}
+
+			// Generate an STXO inclusion proof for each prev
+			// witness.
+			_, stxoProof, err := params.TaprootAssetRoot.Proof(
+				asset.EmptyGenesisID,
+				spentAsset.AssetCommitmentKey(),
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			key := spentAsset.ScriptKey.PubKey.SerializeCompressed()
+
+			stxoInclusionProofs[asset.SerializedKey(key)] =
+				*stxoProof
+		}
+
+		if len(stxoInclusionProofs) == 0 {
+			return nil, fmt.Errorf("no stxo inclusion proofs")
+		}
+
+		proof.InclusionProof.CommitmentProof.STXOProofs =
```

Comment:
Still getting through the PR, but a reminder to also add exclusion proofs: a prevID in a given output needs to prove that it isn't in the STXO for any other output (that has a TAP commitment).